libyul/backends/wasm/EVMToEwasmTranslator.cpp

```diff
@@ -1211,6 +1211,9 @@ function revert(x1, x2, x3, x4, y1, y2, y3, y4) {
 function invalid() {
 	unreachable()
 }
+function memoryguard(x:i64) -> y1, y2, y3, y4 {
```

Comment:
I guess this is okay, but it turns out the wasm dialect doesn't implement objects properly. It is missing the datasize and dataoffset builtins, and only has a non-working version of datacopy in this file.

Reply:
But you still need to add this to the WasmDialect.

Reply:
It isn't part of the wasm dialect, it only needs to be dealt with for EVM to Wasm. I think everything is just fine here :-).

Reply:
(it is needed and occurs during some test case by the way)

Reply:
Ah right memoryguard requires the evm-with-objects, we don't have the same stack issue in wasm.

Reply:
What do you think would make sense to comment where exactly? This is just the implementation of the builtin for the ewasm transform just like for any other as well - and it does the same as for the evm dialect (i.e. just pass through it's argument)...
The memoryguard isn't really related to the stack issue either - we might as well want to reserve memory for some purpose in wasm at some point using the same mechanism under some circumstances.
I'd be happy to add a comment, if you can tell me what exactly and where exactly, because I cannot think of a comment that makes sense here exactly :-).